internals/provider/cluster/resource.go

```diff
@@ -197,6 +199,122 @@ func (r *clusterResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 	}
 }
 
+func ValidateFirewallRuleName(name string) error {
+    validChars := "abcdefghijklmnopqrstuvwxyz0123456789-"
+    for _, char := range name {
+        if !strings.ContainsRune(validChars, char) {
+            return fmt.Errorf("firewall rule name can only contain lowercase letters, numbers, and hyphens, got: %s", name)
+        }
+    }
+
+    if len(name) < 3 || len(name) > 63 {
+        return fmt.Errorf("firewall rule name must be between 3 and 63 characters, got length: %d", len(name))
+    }
+
+    if name[0] < 'a' || name[0] > 'z' {
+        return fmt.Errorf("firewall rule name must start with a lowercase letter, got: %s", name)
+    }
+
+    return nil
+}
+
+func ValidateClusterConfiguration(plan *clusterResourceModel) error {
+    if err := validateFirewallRules(plan.FirewallRules); err != nil {
+        return err
+    }
+
+    if err := validateNetworks(plan.Networks, plan.Nodes, plan.NodeLocation); err != nil {
+        return err
+    }
+
+    return nil
+}
+
+func validateFirewallRules(rules []firewallRuleModel) error {
+    for i, rule := range rules {
+        if err := ValidateFirewallRuleName(rule.Name.ValueString()); err != nil {
+            return fmt.Errorf("invalid firewall rule at index %d: %w", i, err)
+        }
+
+        port := rule.Port.ValueInt64()
+        if port < 1 || port > 65535 {
+            return fmt.Errorf("invalid port number %d at index %d: must be between 1 and 65535", port, i)
+        }
+
+        sources := common.ConvertTFListToStringSlice(rule.Sources)
+        if len(sources) == 0 {
+            return fmt.Errorf("firewall rule at index %d must have at least one source", i)
+        }
+        
+        for _, source := range sources {
+            if !isValidCIDR(source) {
+                return fmt.Errorf("invalid CIDR format %s in firewall rule at index %d", source, i)
+            }
+        }
+    }
+    return nil
+}
+
+func validateNetworks(networks []networkModel, nodes []nodeModel, nodeLocation types.String) error {
```

Comment:
Why are we validating the network here? Isn't this already done by the saas? Can't we just bubble up the errors from there?

Reply:
We are already bubbling up the errors. But we are also implementing some basic validations for some specific fields too. Like we did for the `regions`